tests/thread/thread_stdin.py

```diff
@@ -0,0 +1,42 @@
+# test that having multiple threads block on stdin doesn't cause any issues.
+#
+# The test doesn't expect any input on stdin.
+#
+# This is a regression test for https://github.com/micropython/micropython/issues/15230
+# on rp2, but doubles as a general property to test across all ports.
+import select
```

Comment:
Zephyr has `_thread` but not `select`, so this test fails on zephyr boards with threading enabled.

Probably best to try-except on this import, and skip accordingly (or add an explicit zephyr skip in `run-tests.py`).